airbyte-integrations/connectors/source-close-com/source_close_com/source.py

```diff
@@ -0,0 +1,479 @@
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        if stream_state.get(self.cursor_field):
+            params["date_created__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class OpportunityStatusChangeActivities(IncrementalCloseComStream):
+    cursor_field = "date_created"
+
+    def path(self, **kwargs) -> str:
+        return "activity/status_change/opportunity"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        if stream_state.get(self.cursor_field):
+            params["date_created__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class TaskCompletedActivities(IncrementalCloseComStream):
+    cursor_field = "date_created"
+
+    def path(self, **kwargs) -> str:
+        return "activity/task_completed"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        if stream_state.get(self.cursor_field):
+            params["date_created__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class Events(IncrementalCloseComStream):
+    cursor_field = "date_updated"
+    limit = 50
+
+    def path(self, **kwargs) -> str:
+        return "event"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        if stream_state.get(self.cursor_field):
+            params["date_updated__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class Leads(IncrementalCloseComStream):
+    cursor_field = "date_updated"
+
+    def path(self, **kwargs) -> str:
+        return "lead"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        if stream_state.get(self.cursor_field):
+            params["query"] = f"date_updated > {stream_state.get(self.cursor_field)}"
+        return params
+
+
+class LeadTasks(IncrementalCloseComStream):
+    cursor_field = "date_created"
+    _type = "lead"
+
+    def path(self, **kwargs) -> str:
+        return "task"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["_type"] = self._type
+        if stream_state.get(self.cursor_field):
+            params["date_created__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class IncomingEmailTasks(IncrementalCloseComStream):
+    cursor_field = "date_created"
+    _type = "incoming_email"
```

Comment:
Think about this:
Let's define the new class: CloseComActivitiesMixin(IncrementalCloseComStream), override the request_params with:
```
CloseComActivitiesMixin(IncrementalCloseComStream):

def request_params(...):
  super().request_params(...)

  params["type"] = self._type

  return params
```
then we Inherit all classes where the _type field declared, and remove duplicated request_params.

WDYT?

Reply:
Implemented two mixin classes for Tasks and Activities.